src/commands/annotatedTag.js

```diff
@@ -59,36 +59,27 @@ export async function annotatedTag ({
     }
 
     // Fill in missing arguments with default values
-    if (tagger === undefined) tagger = {}
-    if (tagger.name === undefined) {
-      tagger.name = await config({ fs, gitdir, path: 'user.name' })
-    }
-    if (tagger.email === undefined) {
-      tagger.email = await config({ fs, gitdir, path: 'user.email' })
-    }
+    tagger = { ...tagger }
```

Comment:
Unfortunately, we'll have to use Object.assign here. Too many people ran into issues with object spread. See https://gitter.im/isomorphic-git/Lobby?at=5bd8a0ed435c2a518e2cc779 for background.

Reply:
Alternatively, you could use 
```js
const { name, email, date, timestamp, timezoneOffset } = tagger
```
if those variable names are all available in this scope. Probably can't do that for `commit` since it has both an author and a comitter to worry about... OH but if we put this in a `util` function then that doesn't matter. I kinda like that approach.

Reply:
> Unfortunately, we'll have to use Object.assign here. Too many people ran into issues with object spread. See https://gitter.im/isomorphic-git/Lobby?at=5bd8a0ed435c2a518e2cc779 for background.

I see. I think this kind of terms should be written to `CONTRIBUTING.md` 😭